index.rst

```diff
@@ -4,9 +4,149 @@ File Formats and Layouts for Cell-based Coadds
+Writing to just POSIX filesystems is adequate, as there's no problem with writing to local temporary storage and then uploading separately in our pipeline architecture.
+We do not expect to need the ability to extend existing files.
+
+We expect to compress all image planes with at least lossless compression, and would like to have the capability to perform lossy compression on some planes as well.
+If we do lossy compression, we would quite likely want to do our own quantization (our `afw` library already has code for this that we prefer over CFITSIO's).
+DESC has expressed an interest in applying lossy compression to any coadd files they transfer to NERSC, if we have not done so already.
+
+Constraints from FITS
+=====================
+
+Because we need to be able to read subimages efficiently, file-level compression is not an option: our only options are FITS image "tile compression" and perhaps lesser-known FITS 4.0's binary table compression.
+Combined with our need to subimage reads over object stores, this already puts us beyond what third-party FITS libraries can do:
+
+- CFITSIO can do image tile compression, including efficient subimage reads that take advantage of the tile structure, but only on POSIX filesystems (or blocks of contiguous memory that correspond to the on-disk layout).
+  It can also do subset reads of FITS binary tables, again only on POSIX filesystems.
+  This imposes the same limitation on its Python bindings in the ``fitsio`` package.
+
+- Astropy can do image tile compression with efficient subimage reads on POSIX filesystems, and it can do uncompressed reads (including efficient subimage reads) against both POSIX and object stores via `fsspec`, but it delegates to vendored CFITSIO code for its tile compression code and does not support the combination of these two.
+  Astropy cannot do subset reads of FITS binary tables, even on POSIX filesystems.
+
+- I am not aware of any library that can do FITS binary table compression at all.
+
+As a result, I expect us to need to write some low-level code of our own to do subimage reads over object stores, though this is not specific to cell-based coadds: we need this for all of our image data products.
+Whether to contribute more general code upstream (probably to Astropy) or focus only on reading the specific files we write ourselves is an important open question; the general solution would be very useful the community, but its scope is unquestionably larger.
+
+Finally, the WCS of any single cell (or the inner-cell stitched image) will be simple and can be exactly represented in FITS headers (unlike the WCSs of our single-epoch images).
+An additional FITS WCS could be used to describe the position of a single-cell image within a larger grid.
```

Comment:
This position is simply a pixel location and the dimension of the image.

Reply:
I think Jim is referring to the Rubin convention of having an "alternative WCS" (we usually use WCS `A`) that provides the mapping from local FITS 1-based pixel coordinates to tract-level 0-based coordinates.